### DISCUSSION TITLE
How do I modify the ID of the root node， Name of hydration data

### DISCUSSION THREAD

**Discussion by parlay96** [vercel/next.js#58526] [Ideas]
### Goals

![image](https://github.com/vercel/next.js/assets/32978177/1bfcad0e-8308-4fe3-8e4e-7b2b6f29e5cb)

### Background

Our company does not want others to directly see that we have used the next technology stack, as requested by our leaders

### Proposal

Can I modify it？

**Comment by Dev1OptimusTecno** [comment]
you can use the new app router and create your own layout that does not have id="_next" , the _next_data_ could be replaced by server-components , about the /_next prefix have a look at this issue: 
https://github.com/vercel/next.js/issues/5602

**Reply by parlay96** [reply]
Thank you

**Comment by willfolsom** [comment]
Grep through all your built files and replace _next with something else, i.e.

`grep -rli '_next' * | xargs -I@ sed -i '' 's/_next/next2/g' @`

In this example I want to use `next2` instead of `_next`.

You'd probably also need to rename the _next/ directory to your new naming as well.

Take this with a grain of salt but it works for me. Probably worth noting I build to static.

**Reply by parlay96** [reply]
Thank you
